doc/guides/extending-inspector-protocol.md

```diff
@@ -0,0 +1,77 @@
+# Extending Node.js Inspector Protocol
+
+Node.js Inspector Protocol (based on the [Chrome DevTools](https://chromedevtools.github.io/devtools-protocol/v8/))
+is designed to be easily extensible by the implementers. This document provides
+details on such extension.
+
+## Terminology
+
+* *Inspector session* - message exchange between protocol client and Node.js
+* *Protocol handler* - object with a liftime of the inspector session that
+interacts with the protocol client.
+* *Protocol domain* - 'API unit' of the inspector protocol. Defines message
+formats for specific and related functionality. Node.js exposes protocol
+domains provided by the V8. These include `Runtime`, `Debugger`, `Profiler` and
+`HeapProfiler`.
+* *Message* - a JSON string passed between Node.js backend and a inspector
+protocol client. Message types are: request, response and notification. Protocol
+client only sends out requests. Messages are asynchronous and client should not
+assume response will be sent immidiately after serving the request.
+* *Method field* - a mandatory field in request and notification messages.
+This field value includes *domain* and *method* strings separated by a dot
+(e.g. `Debugger.pause`).
+
+## API
+
+`internal/inspector_node_domains` module provides API for extending inspector
+protocol exposed by the Node.js
+
+### inspector_node_domains.registerDomainHandlerClass(domain, constructor)
+
+* domain {string} - a JavaScript identifier string that serves as a unique
+identifier for the messages that target this domain. Domains should be unique.
+It is not permitted to override built-in Node.js domains.
+* constructor {ES6 class|constructor function} - creates a protocol handler that
```

Comment:
Just a reminder that both `ES6 class` and `constructor function` need to be added in [type-parser.js](https://github.com/nodejs/node/blob/master/tools/doc/type-parser.js)

Reply:
fwiw this isn’t part of the external API documentation